crates/rustc_codegen_spirv/src/spirv_type.rs

```diff
@@ -305,50 +305,43 @@ impl SpirvType {
 
     pub fn sizeof<'tcx>(&self, cx: &CodegenCx<'tcx>) -> Option<Size> {
         let result = match *self {
-            Self::Void => Size::ZERO,
+            Self::Void | Self::Opaque { .. } => Size::ZERO,
             Self::Bool => Size::from_bytes(1),
-            Self::Integer(width, _) => Size::from_bits(width),
-            Self::Float(width) => Size::from_bits(width),
+            Self::Integer(width, _) | Self::Float(width) => Size::from_bits(width),
             Self::Adt { size, .. } => size?,
-            Self::Opaque { .. } => Size::ZERO,
             Self::Vector { element, count } => {
                 cx.lookup_type(element).sizeof(cx)? * count.next_power_of_two() as u64
             }
             Self::Array { element, count } => {
                 cx.lookup_type(element).sizeof(cx)? * cx.builder.lookup_const_u64(count).unwrap()
             }
             Self::RuntimeArray { .. } => return None,
-            Self::Pointer { .. } => cx.tcx.data_layout.pointer_size,
-            Self::Function { .. } => cx.tcx.data_layout.pointer_size,
```

Comment:
This one is wrong I'm pretty sure, a SPIR-V function is not a function pointer, and should `return None` here - assuming that still can compile examples and tests, ofc.

Reply:
nice that the lint caught it, it does read better and easier to spot bugs when same arms are collapsed!

can you do an issue or fix it in a separate PR? not sure if good idea to fix and change behavior here in the lint PR